torch/utils/cpp_extension.py

```diff
@@ -778,7 +790,29 @@ def _check_cuda_version(self):
                     if getattr(cuda_ver, "major", float("nan")) != getattr(torch_cuda_version, "major", float("nan")):
                         raise RuntimeError(CUDA_MISMATCH_MESSAGE.format(cuda_str_version, torch.version.cuda))
                     warnings.warn(CUDA_MISMATCH_WARN.format(cuda_str_version, torch.version.cuda))
-
+                if sys.platform.startswith('linux'):
```

Comment:
This assumes, that `gcc` is only compiler used on linux, which is obviously not the case, when `clang` is used